# Additional File 2
This github repository includes R software codes used for a study on the evaluation of 14 differential gene expression methods for RNA-sequencing data (particularly for mRNA and lncRNA). Fourteen popular DGE tools are considered in our study. All considered tools are popular (in terms of number of citation) and available on R software packages. All of them start from gene or transcript level read counts. Most methods were applied using their default settings, except for DESeq2 and PoissonSeq which required certain settings to be changed to avoid filtering out too many low abundant genes (such as lncRNA). In particular, in DESeq2 the independent filtering was disabled in the results() function, and in PoissonSeq the filter cutoffs for the sum and average counts of genes across samples were set to 1 and 0.01, respectively. All DE tools were applied at the nominal 5% FDR level, unless mentioned otherwise. All computations were performed in R version 3.3.2. baySeq was not included in the simulation study due to its slow computation time.

The study has three main modules: (1) comparing selected normalization methods, (2) concordance analysis of 14 DGE analysis tools using 6 publicly accessible real RNA-seq data, and (3) non-parametric simulation study to evaluate 13 tools with respect to false discovery rate and sensitivity.

Five normalization methods that are used (as default) in conjunction to DGE tools (targeted tools in our study) are compared using two qualitative and one quantitative metrics: their capability to reduce variability that attributes to technical sources, their capability of eliminating bias due to library size differences, and their effect on DE analysis with the moderated t-test of the limma package. The normalization methods are  quantile normalization (QN) implemented in limma (limmaQN), Trimmed Mean of M-values  implemented in edgeR, limma (limmaVoom and limmaVoom+QW), baySeq, and QuasiSeq, Medians of Ratios (MR) implemented in DESeq, DESeq2, and limma (limmaVst), the goodness-of-fit statistics approach in PoissonSeq, and the re-sampling technique implemented in SAMSeq. Codes used for this module are compiled in a folder "comparing-normalization-methods".

